library/src/main/java/com/novoda/downloadmanager/RoomAppDatabase.java

```diff
@@ -1,12 +1,31 @@
 package com.novoda.downloadmanager;
 
 import android.arch.persistence.room.Database;
+import android.arch.persistence.room.Room;
 import android.arch.persistence.room.RoomDatabase;
+import android.content.Context;
 
 @Database(entities = {RoomBatch.class, RoomFile.class}, version = 1)
 abstract class RoomAppDatabase extends RoomDatabase {
 
+    private static volatile RoomAppDatabase SINGLE_INSTANCE;
```

Comment:
@zegnus couldn't use the inner "holder" class to make this thread safe as it depends on `Context`. This should do the job though.